markovbot/commands.py

```diff
@@ -44,3 +47,71 @@
     await markovbot.say(sentence)
 
 
+@markovbot.command(pass_context=True, help='Mock the last specified user message.')
+async def mock(context, user: str = None):
+    server_context = context.server_context
+
+    if not server_context.is_ready:
+        await markovbot.say('I am still learning from all your messages. Try again later.')
+        return
+
+    if user is None:
+        log.debug('Bot issued say command on server %s',
+                  server_context.server.name)
+        sentence = server_context.markov.make_sentence_server()
+
+        if sentence is None:
+            await markovbot.say('Unable to generate message. This is probably due to a lack of messages on the server.')
+            return
+
+    else:
```

Comment:
I dont think you need the stuff that is basically just checking if the server is initialized yet, as you arnt doing anything with those services. Granted this stuff should really be taken care of for you and not exposed at the command layer anyway, but I was lazy when I did that. 

I think you just need what is after the else block.

Just let me check that I understand what you are trying to do:
1. A user issues the command:  !prefix mock "User name"
2. Get the users last message from the channel if they specified the user, else just get the last message.
3. Call the mock string function on the sentence, and say it back to the server.

Reply:
I removed the logic that you think isn't necessary. I thought we might still want to check whether the server is ready or not.

`    if not server_context.is_ready:
        await markovbot.say('I am still learning from all your messages. Try again later.')
        return`